pkg/controller/actuator.go

```diff
@@ -374,35 +374,27 @@ func (a *actuator) createSeedResources(
 ) error {
 	var err error
 
+	alwaysAllowedCIDRs = append(alwaysAllowedCIDRs, shootSpecificCIRDs...)
+
 	apiEnvoyFilterSpec, err := envoyfilters.BuildAPIEnvoyFilterSpecForHelmChart(
-		spec.Rule, hosts, append(alwaysAllowedCIDRs, shootSpecificCIRDs...), istioLabels,
+		spec.Rule, hosts, alwaysAllowedCIDRs, istioLabels,
 	)
 	if err != nil {
 		return err
 	}
 
-	ingressSNI := helper.GetSeedIngressDomain(cluster.Seed)
-	shootID := helper.ComputeShootID(cluster.Shoot)
-
-	defaultIstioLabels := map[string]string{
-		"app":   "istio-ingressgateway",
-		"istio": "ingressgateway",
+	cfg := map[string]interface{}{
+		"shootName":          cluster.Shoot.Status.TechnicalID,
+		"targetNamespace":    istioNamespace,
+		"apiEnvoyFilterSpec": apiEnvoyFilterSpec,
 	}
 
-	ingressEnvoyFilterSpec, err := envoyfilters.BuildIngressEnvoyFilterSpecForHelmChart(
-		spec.Rule, ingressSNI, shootID, append(alwaysAllowedCIDRs, shootSpecificCIRDs...), defaultIstioLabels,
-	)
+	err = envoyfilters.BuildIngressEnvoyFilterSpecForHelmChart(
+		cluster, spec.Rule, alwaysAllowedCIDRs, cfg)
 	if err != nil {
 		return err
 	}
 
-	cfg := map[string]interface{}{
-		"shootName":              cluster.Shoot.Status.TechnicalID,
-		"targetNamespace":        istioNamespace,
-		"apiEnvoyFilterSpec":     apiEnvoyFilterSpec,
-		"ingressEnvoyFilterSpec": ingressEnvoyFilterSpec,
```

Comment:
I like the previous approach better: let's keep calculating the sub-values in the function and then pass it to `cfg["ingressEnvoyFilterSpec"]` similar to how `BuildAPIEnvoyFilterSpecForHelmChart` works.